p2p/net/swarm/swarm_event_test.go

```diff
@@ -0,0 +1,66 @@
+package swarm_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/event"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/p2p/host/eventbus"
+	. "github.com/libp2p/go-libp2p/p2p/net/swarm"
+	swarmt "github.com/libp2p/go-libp2p/p2p/net/swarm/testing"
+
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/stretchr/testify/require"
+)
+
+func newSwarmWithSubscription(t *testing.T) (*Swarm, event.Subscription) {
+	t.Helper()
+	bus := eventbus.NewBus()
+	sw := swarmt.GenSwarm(t, swarmt.EventBus(bus))
+	t.Cleanup(func() { sw.Close() })
+	sub, err := bus.Subscribe(new(event.EvtPeerConnectednessChanged))
+	require.NoError(t, err)
+	t.Cleanup(func() { sub.Close() })
+	return sw, sub
+}
+
+func checkEvent(t *testing.T, sub event.Subscription, expected event.EvtPeerConnectednessChanged) {
+	t.Helper()
+	select {
+	case ev, ok := <-sub.Out():
+		require.True(t, ok)
+		evt := ev.(event.EvtPeerConnectednessChanged)
+		require.Equal(t, expected.Connectedness, evt.Connectedness, "wrong connectedness state")
+		require.Equal(t, expected.Peer, evt.Peer)
+	case <-time.After(time.Second):
+		t.Fatal("didn't get PeerConnectedness event")
+	}
+
+	// check that there are no more events
+	select {
+	case <-sub.Out():
+		t.Fatal("didn't expect any more events")
+	case <-time.After(50 * time.Millisecond):
```

Comment:
probably bump this to 100ms for CI. I know it'll slow things down 50ms, sorry.

Reply:
Done.